docs/concepts/cluster_queue.md

```diff
@@ -0,0 +1,24 @@
+# Cluster Queue
+
+A `ClusterQueue` is a cluster-scoped resource that governs a pool of resources
+such as CPU, memory and hardware accelerators. A `ClusterQueue` defines:
+- The resource _flavors_ that it manages, with usage limits and order of consumption.
+- Fair sharing rules across the tenants of the cluster.
+
+Only cluster administrators should create `ClusterQueue` objects.
```

Comment:
lets also add something about personas (Batch Admin and Batch User), perhaps in glossary too

Reply:
I was thinking about that too. I'll do it in a follow up.